### ISSUE TITLE
Implement loadable shift register

### ISSUE THREAD

**Issue by zhemao-openai** [xlsynth/bedrock-rtl#159]
We have the delay modules that can take serial input. What's missing is a shift register module that can be loaded in parallel. I imagine an interface something like this:

```
module br_shift_reg #(
  parameter int BitWidth = 1,
  parameter int NumStages = 2
) (
  input logic clk,
  input logic rst,

  input logic reinit,
  // value gets set to initial_value on reset or when reinit is asserted
  input  logic [NumStages-1:0][BitWidth-1:0] initial_value,
  output logic [NumStages-1:0][BitWidth-1:0] value,

  // when shift_en is 1, value becomes {value[NumStages-2:0], shift_in}
  input logic shift_en,
  input  logic [BitWidth-1:0] shift_in,
  // shift_out is the same as value[NumStages-1]
  output logic [BitWidth-1:0] shift_out
);
```

**Comment by mgottscho** [comment]
lgtm

**Comment by zhemao-openai** [comment]
This was fixed by PR#189
